core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -65,6 +64,7 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     Metric.runtime.fiberForkLocations.unsafe.update(fiberId.location.toString, tags)(Unsafe.unsafe)
   }
 
+  @volatile private var observers  = Nil: List[Exit[E, A] => Unit]
```

Comment:
I think let's change it back to having it as non-volatile. Volatile vars require extra memory and they're slower to initialize. Since we need FiberRuntime to be as lightweight as humanly possible, let's revert this change and we can revisit it if issues come up again